### PR TITLE
Fixed many warnings

### PR DESCRIPTION
**Pull request by NickNackGus**

### CODE REVIEW

**Comment by NickNackGus** [comment]
And to balance out the warnings that were fixed, here's a new warning: this needs to be tested still.

**Comment by NickNackGus** [comment]
...also the reviewdog needs to be updated to handle compiling with gradle.

File: src/main/java/com/playmonumenta/structures/StructuresAPI.java [hunk 1]
```diff
@@ -697,24 +682,21 @@ private static void ensureTask(Plugin plugin) {
...
+
+					try {
+						Bukkit.getScheduler().runTask(plugin, task.mStartTask);
+						task.mSignal.get(45, TimeUnit.SECONDS);
+					} catch (Exception ex) {
+						plugin.getLogger().severe("Structure task took longer than 45s to complete! Continuing to the next task. Exception: " + ex.getMessage());
+					}
 				}
 			}
 		};
```
Comment (Combustible) [inline src/main/java/com/playmonumenta/structures/StructuresAPI.java:702]: I think this whole thing needs some restructuring, really weird setup right now. Need some timeout for the PENDING_TASKS.take() I think? 

Maybe we should just revert this entire block and move this change to another PR

File: src/main/java/com/playmonumenta/structures/managers/EventListener.java [hunk 2]
```diff
@@ -21,20 +21,14 @@
...
 		SpawnReason.VILLAGE_DEFENSE,
 		SpawnReason.VILLAGE_INVASION
 	);
 
-	StructuresPlugin mPlugin = null;
-
-	public EventListener(StructuresPlugin plugin) {
-		mPlugin = plugin;
-	}
+	public EventListener() {}
```
Comment (Combustible) [inline src/main/java/com/playmonumenta/structures/managers/EventListener.java:31]: this line doesn't need to exist

File: src/main/java/com/playmonumenta/structures/managers/RespawningStructure.java [hunk 3]
```diff
@@ -222,11 +229,18 @@ public static CompletableFuture<RespawningStructure> withParameters(StructuresPl
...
+	                            int extraRadius,
+		                        String configLabel,
+		                        String name,
+		                        List<String> genericPaths,
+		                        Vector loadPos,
+		                        int respawnTime,
+		                        int ticksLeft,
+		                        @Nullable String postRespawnCommand,
+		                        @Nullable String nextRespawnPath,
+		                        @Nullable SpawnerBreakTrigger spawnerBreakTrigger) {
```
Comment (Combustible) [inline src/main/java/com/playmonumenta/structures/managers/RespawningStructure.java:243]: weird indentation

File: src/main/java/com/playmonumenta/structures/managers/RespawningStructure.java [hunk 4]
```diff
@@ -342,11 +354,10 @@ public void forcedRespawn(Player player, boolean ignoreDefaultSpawnerCount) {
...
 					} else {
 						int minutesLeft = mTicksLeft / (60 * 20);
-						String minutesPlural = (minutesLeft > 1) ? "s" : "";
 						for (Player p : mWorld.getPlayers()) {
 							if (isNearby(p)) {
 								p.sendMessage(ChatColor.RED + "" + ChatColor.BOLD + mName + " will forcibly respawn in "
-										+ minutesLeft + " minute" + minutesPlural + "!");
+										+ minutesLeft + " minutes!");
 							}
 						}
```
Comment (Combustible) [inline src/main/java/com/playmonumenta/structures/managers/RespawningStructure.java:362]: I think this was still needed? "1 minutes" is clunky

Comment (NickNackGus) [reply src/main/java/com/playmonumenta/structures/managers/RespawningStructure.java:362]: According to IntelliJ it's not used, but I was thinking the same thing. It turns out this code path is never reached with less than 2 minutes remaining.

Comment (Combustible) [reply src/main/java/com/playmonumenta/structures/managers/RespawningStructure.java:362]: Oh, I see, cool. That's interesting that the IDE caught that.

**Review by Combustible — APPROVED** [review]
